torch/jit/__init__.py

```diff
@@ -2059,7 +2060,10 @@ def _get_named_tuple_properties(obj):
 
 def _create_named_tuple(t, unqual_name, field_names):
     TupleType = collections.namedtuple(unqual_name, field_names)
-    return TupleType(*t)
+    setattr(torch.jit._tuples, unqual_name, TupleType)
```

Comment:
btw, it looks like we're creating a named tuple each time from scratch here: https://github.com/pytorch/pytorch/blob/ed60d9fcf9fa0205a1e6872c89df5b37c23ada50/torch/csrc/jit/pybind_utils.h#L563

shouldn't we instead just retrieve an existing one if it already exists?

Reply:
No, we explicitly do not do that because we're structurally typing NamedTuple. It doesn't make sense to do that in light of serialization

Reply:
Wait, I thought we call it every time when namedtuple passes from c++ to python. Am I missing something?

Reply:
```
# foo1.py
import torch
from typing import NamedTuple

class MyNT(NamedTuple):
    i : int
    b : float

@torch.jit.script
def foo():
    return MyNT(3, 3.5)

torch.jit.save(foo, 'foo.zip')
```

```
# foo2.py
import torch

loaded = torch.jit.load('foo.zip')

print(loaded())
```

When running the second script, the original class doesn't exist in the Python interpreter, so we can't retrieve it to construct it